code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm

```diff
@@ -54,6 +63,10 @@
 		return
 	if(Adjacent(atom_to_walk_to))
 		return
+	if(!isliving(atom_to_walk_to))
+		mob_parent.face_atom(atom_to_walk_to)
+		mob_parent.UnarmedAttack(atom_to_walk_to, mob_parent)
+		return
```

Comment:
It would be better to have an early return here for if(isliving(atom_to_walk_to)), and that conscious check. At the moment you're duplicating the face_atom and attack code for no real reason.